Vic2ToHoI4/Source/HOI4World/HoI4World.cpp

```diff
@@ -1669,3 +1682,77 @@ closedProvinces.insert(make_pair(thisAdjacency.to, thisAdjacency.to));
 return -1;
 
 }*/
+
+void HoI4::World::setSphereLeaders()
+{
+	for (auto GP: greatPowers)
+	{
+		for (auto& relationItr: GP->getRelations())
+		{
+			if (relationItr.second.getSphereLeader())
+			{
+				auto sphereling = findCountry(relationItr.first);
+				sphereling->setSphereLeader(GP->getTag());
+			}
+		}
+	}
+}
+
+void HoI4::World::processInfluence()
+{
+	for (auto GP: greatPowers)
+	{
+		for (auto& relationItr: GP->getRelations())
+		{
+			auto influencedCountry = findCountry(relationItr.first);
+			bool isFriendlyCountry = relationItr.second.getGuarantee();
+			auto influenceValue = relationItr.second.getInfluenceValue();
+
+			if (isFriendlyCountry)
+			{
+				influencedCountry->addGPInfluence(GP->getTag(), influenceValue);
+			}
+		}
+	}
+}
+
+void HoI4::World::determineSpherelings()
+{
+	for (auto& GP: greatPowers)
+	{
+		for (auto relationItr: GP->getRelations())
+		{
+			bool isInSphere = relationItr.second.getSphereLeader();
+			bool notPuppet = (GP->getPuppets().find(relationItr.first) == GP->getPuppets().end());
+			auto allies = GP->getAllies();
+			bool isAlly = (allies.find(relationItr.first), allies.find(relationItr.first) != allies.end());
```

Comment:
You can use allies.count(relationItr.first) instead (any non-zero will count as true).

Reply:
I'm not sure if I understand correctly. `relationItr.first` is the tag of the country I'm trying to check if it's listed as an ally in GP's relations.

Reply:
Got it. No need for this long an expression, when count() returns the same value `true`